src/Firestore/Transaction.php

```diff
@@ -0,0 +1,330 @@
+<?php
+/**
+ * Copyright 2017 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Google\Cloud\Firestore;
+
+use Google\Cloud\Core\DebugInfoTrait;
+use Google\Cloud\Core\Exception\NotFoundException;
+use Google\Cloud\Firestore\Connection\ConnectionInterface;
+
+/**
+ * Represents a Firestore transaction.
+ *
+ * This class should be accessed inside a transaction callable, obtained via
+ * {@see Google\Cloud\Firestore\FirestoreClient::runTransaction()}.
+ *
+ * Note that method examples, while shown as being called directly for the sake
+ * of brevity, should be called only within the context of a transaction
+ * callable, as noted above.
+ *
+ * Example:
+ * ```
+ * use Google\Cloud\Firestore\FirestoreClient;
+ * use Google\Cloud\Firestore\Transaction;
+ *
+ * $firestore = new FirestoreClient();
+ * $document = $firestore->document('users/john');
+ * $firestore->runTransaction(function (Transaction $transaction) use ($document) {
+ *     // Manage Transaction.
+ * });
+ * ```
+ */
+class Transaction
+{
+    use SnapshotTrait;
+    use DebugInfoTrait;
+
+    /**
+     * @var ConnectionInterface
+     */
+    private $connection;
+
+    /**
+     * @var ValueMapper
+     */
+    private $valueMapper;
+
+    /**
+     * @var string
+     */
+    private $transaction;
+
+    /**
+     * @var string
+     */
+    private $database;
+
+    /**
+     * @var WriteBatch
+     */
+    private $writer;
+
+    /**
+     * @param ConnectionInterface $connection A connection to Cloud Firestore.
+     * @param ValueMapper $valueMapper A Firestore Value Mapper.
+     * @param string $database The database name.
+     * @param string $transaction The transaction ID.
+     */
+    public function __construct(
+        ConnectionInterface $connection,
+        ValueMapper $valueMapper,
+        $database,
+        $transaction
+    ) {
+        $this->connection = $connection;
+        $this->valueMapper = $valueMapper;
+        $this->database = $database;
+        $this->transaction = $transaction;
+
+        $this->writer = new WriteBatch($connection, $valueMapper, $database, $transaction);
+    }
+
+    /**
+     * Get a Document Snapshot.
+     *
+     * Unlike {@see Google\Cloud\Firestore\Document::snapshot()}, if the document
+     * does not exist, this method will throw
+     * {@see Google\Cloud\Core\Exception\NotFoundException}.
```

Comment:
What is the reason why it is different?

Reply:
The Firestore Server API implementation doc says "The get(documentReference) method returns a DocumentSnapshot or an error if no document exists at documentReference."

Reply:
Sorry, this is an issue with the API doc then. I fixed the document. I hope this didn't cause too much extra work.

Reply:
fixed!